test/model-helpers/UserList.test.js

```diff
@@ -5,26 +5,17 @@ import {
   setThing,
   getSolidDataset,
   saveSolidDatasetAt,
-  saveAclFor
+  saveAclFor,
+  addMockResourceAclTo
 } from '@inrupt/solid-client';
 import { expect, vi, it, describe, afterEach, beforeEach } from 'vitest';
 import { loadUserList, addUser, removeUser, makeUserIntoThing } from '../../src/model-helpers';
 import { RDF_PREDICATES } from '../../src/constants';
 
-vi.mock('@inrupt/solid-client', async () => {
-  const actual = await vi.importActual('@inrupt/solid-client');
-  return {
-    ...actual,
-    saveAclFor: vi.fn(),
-    saveSolidDatasetAt: vi.fn(),
-    getSolidDataset: vi.fn((url) => Promise.resolve(mockSolidDatasetFrom(url))),
-    setAgentResourceAccess: vi.fn(),
-    setAgentDefaultAccess: vi.fn()
-  };
-});
+vi.mock('@inrupt/solid-client');
 
 const makeMockUserListDataset = (userList, url) => {
-  let dataset = mockSolidDatasetFrom(url);
+  let dataset = addMockResourceAclTo(mockSolidDatasetFrom(url));
```

Comment:
We can build valid solid datasets with this pattern.  The `setAcl` functions error if we pass them a mock dataset that didn't have `addMockResourceAclTo` called on it.